### PR TITLE
Connect AOIStreamingMonitor to PostgreSQL database

### PR DESCRIPTION
**Pull request by CloudNiner**
# Overview

This PR adds the methods necessary to retrieve all AOIs along with their supplementary information necessary to send emails via postgres connection. The intent here is to present, have a discussion about, and settle on a return type for `queryAreaOfInterests` that supports the join operations necessary to combine this return type with the AugmentedDiff stream in some way.

## TODO:

- [x] Filter AOIs in the postgres query to only include ones configured for the desired notification interval -- this shouldn't change the type of the response though

## Notes (from commit message)

Switched to a Config object to pull all postgres
configuration from environment variables. Each
part of the connection is its own variable so that
its easier to view, configure and diff individual
parts of the connection string, learned the hard
way in other projects.

Added in the methods that get and set the sequence
value for the desired interval.

Switched Interval to a sealed trait and case object
for easier matching and type safety.

An automated scalafmt snuck its way in there in the
middle of my efforts. Untangling it was a bit of a
headache so I just left it, even though it should
be a separate commit.

### CODE REVIEW

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:84]: With the non-compiling section below commented out, I ran the following:
```
 test:runMain osmesa.analytics.oneoffs.StreamingAOIMonitor --interval w --augmented-diff-source s3://<bucket>/augmented-diffs
```
and got:
```
[info] AOIData(2756a4a0-08ce-46de-aee9-8959a2548777,0587df26-28b5-4374-a9ea-8a4cf2de2d3f,2019-10-08 17:20:34.574671,POLYGON ((-67.32421875 9.1
78024574578373, -65.511474609375 9.178024574578373, -65.511474609375 10.746969318460001, -67.32421875 10.746969318460001, -67.32421875 9.17802
4574578373)),Caracas,foo@example.com) 
```
printed from this line which connected to the database holding this information.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:190]: This call is new.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:288]: All of the significant changes for this PR are from this line down to the bottom of the file.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: This was orginally `ST_AsText` to pull the geoms as WKT, but WKB should be far more efficient.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:412]: Removed the `--database-url` CLI argument in favor of this. See commit log for reasoning.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (jpolchlo) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:35]: Moving this to a separate line and reindenting made this diff very hard to read.  These types of cosmetic changes should probably be put off until after substantive comments are addressed in the future.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (jpolchlo) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: This should probably send a comprehensive error message to the logger since this will be the only way we have to identify a problem with the checkpoint setting.  I.e., in case there is a problem with the run and we have to go back and manually update the checkpoint table, we should know what the sequence number is and for which interval (mostly to minimize the possibility of a screw-up).

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (jpolchlo) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:284]: I want to point out that this function wraps DB accesses in a spark action.  Depending on the environment, the master node may not have the JDBC drivers properly installed, but the executors will.  I've surely encountered this problem on EMR, but I can't recall if such weirdness is also present on ECS.  Keep this in mind when we actually deploy, in case DB accesses start failing.  This will be the pattern to fall back on.

**Review by jpolchlo — APPROVED** [review]
LGTM

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [reply src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:35]: Yeah, I made note of this in the commit log and PR description. Totally understood. I couldn't untangle it from in flight commits all that well so I left it in. Don't plan to let it happen in the future.

> An automated scalafmt snuck its way in there in the
middle of my efforts. Untangling it was a bit of a
headache so I just left it, even though it should
be a separate commit.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [reply src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: 👍 I'll swap back to `warnMessage` before merging

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala
Comment (CloudNiner) [reply src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala:284]: Huh. Well that makes sense but I would never have guessed that's why. I assumed that you had just copy pasted these two functions to stub them in without addressing implementation specifics. 

If this becomes an issue, won't it also then be an issue for the `queryAreaOfInterest` call that gets the AOIs from the database?

In any case, I don't plan to make any immediate changes here (aside from perhaps a comment) but noted for later for sure.

**Comment by CloudNiner** [comment]
@jpolchlo this is ready to merge, which I cannot do since I don't have write access on your repo. Can you merge if there's no outstanding change requests here?

**Pull request merged**
